### DISCUSSION TITLE
Flashing process automatically expands APP partition

### DISCUSSION THREAD

**Discussion by CHNtentes** [OE4T/meta-tegra#1296] [Q&A] [answered]
Hi.

I found that even though I set `allocation_attribute` to "0x8", and a fixed size (40GB) for my APP partition, the flashing script always expands it to the end, taking up all remaining space. This issue only applies to APP (/dev/nvme0n1p1) partition.

I'd like the size to be the same as I set (I'm aware that some space is left unused) . Is there a method to change this behaviour? 

Thanks in advance.

Here is my partition table: [initrd-flash.xml.txt](https://github.com/OE4T/meta-tegra/files/12067053/initrd-flash.xml.txt)

Flashing process screenshot:
![image](https://github.com/OE4T/meta-tegra/assets/99521008/5ae0a766-8449-46de-8701-9206d7366e58)

**Comment by madisongh** [comment, marked as answer]
To work around this feature, you could add another partition to the end of your layout with the fill-to-end attribute set on it.

Otherwise, a patch to the `initrd-flash.sh` script would be needed to pass the `--no-final-part` flag to the `make-sdcard` script when writing the partitions to the external storage device

**Reply by CHNtentes** [reply]
Modified initrd-flash.sh and the issue was solved. Thanks again for your help.
